Added documentation for Airflow Upgrade Check

vikramkoka opened the pull request:
Added a documentation file for the Airflow Upgrade Check scripts. This is still
fairly early, but wanted a starting point to detail how to get the python
package, run the script, understand and apply the results.

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

vikramkoka committed:
Added documentation for Aiflow Upgrade Check
Added a documentation file for the Airflow Upgrade Check scripts. This is still
fairly early, but wanted a starting point to detail how to get the python
package, run the script, understand and apply the results.

mik-laj commented on docs/apache-airflow/upgrade-check.rst:
```diff
@@ -0,0 +1,145 @@
+ .. Licensed to the Apache Software Foundation (ASF) under one
```
I think this should be moved to a separate documentation package so that this documentation can be updated independently of Airflow. Simply put, you need to create a new directory in / docs / and then update the /docs/conf.py and /docs/build_docs.py files so that the package can be built.

ashb replied:
If we do this how can we link to it this from a (yet-to-be-created) `updating-to-2-0.rst` in `docs/apache-airflow/`?

ashb replied:
Oh -- but also, maybe slight complication: the upgrade/check code doesn't exist on master, just on v1-10-* branches.

mik-laj replied:
Documentation for airflow upgrade-check, we can publish from the master page, if we do not generate any pages automatically.

Docs about cross-reference syntax: https://github.com/apache/airflow/tree/master/docs#cross-referencing-syntax

kaxil commented on docs/apache-airflow/upgrade-check.rst:
```diff
@@ -0,0 +1,145 @@
+
+The second problem was identified in one of the DAGs. In this case, this import
+statement for the PythonOperator needs to be changed, since the location is different
+in Airflow 2.0. We will discuss how in the next section.
```
```suggestion
Understanding what is being checked
'''''''''''''''''''''''''''''''''''

The Upgrade Check checks the configuration data from airflow.cfg, the metadata from the Airflow
database, as well as the DAGs which have been set up in the current Airflow environment.

Using the above results as an example, there are two specific problems which have
been identified.

The first problem is identified in the configuration file airflow.cfg where the current configuration
option for the fernet_key is no longer acceptable and needs to be changed.

The second problem was identified in one of the DAGs. In this case, this import
statement for the PythonOperator needs to be changed, since the location is different
in Airflow 2.0. We will discuss how in the next section.
```

vikramkoka committed:
Moved upgrade-check doc file to a new docs directory
Moved the upgrade-check.rst file to a new docs sub-directory based on
feedback from Kamil. Also added a bit more content to it around fixing identified
problems.

kaxil commented on docs/upgprade-check/upgrade-check.rst:
```diff
@@ -0,0 +1,161 @@
+as well as make it compatible for Airflow 2.0.
+
+.. code-block:: python
+    from airflow.operators import PythonOperator
```
This does not work with 1.10.14 does it @vikramkoka ?

vikramkoka replied:
Actually, this does seem to work in 1.10.14rc1

vikramkoka replied:
I will retest with 1.10.14rc4 and 2.0beta3 in just a bit to make sure.

vikramkoka replied:
Drats! This does work in 1.10.14rc4, but it does NOT work in 2.0.0b3. 
I will fix it in the doc, but not right now, I don't know of a way where it works in both.

mik-laj replied:
See: https://github.com/apache/airflow/commit/0e5eee83b14b2a57345370b14e91404d518f0bf4#diff-4169e001fd63ad284011b12a578f74ae09ebeda1c5318256c0d7a97633c738c8

vikramkoka committed:
Fixed error in Python operator recommendation
Fixed the error in the description of how to handle problems detected
in the upgrade check with respect to importing the Python Operator.

mik-laj commented on docs/upgprade-check/upgrade-check.rst:
```diff
@@ -0,0 +1,163 @@
+
+    Problems:
+
+    1.  Using airflow.operators.python_operator.PythonOperator will be replaced by airflow.operators.python.PythonOperator. Affected file:
```
```suggestion
    1.  Using ``airflow.operators.python_operator.PythonOperator`` will be replaced by ``airflow.operators.python.PythonOperator``. Affected file:
```

mik-laj commented on docs/upgprade-check/upgrade-check.rst:
```diff
@@ -0,0 +1,163 @@
+    fernet_key =
+
+The second problem was identified in one of the DAGs. In this case, this import
+statement for the PythonOperator needs to be changed, since the location is different
```
```suggestion
statement for the ``PythonOperator`` needs to be changed, since the location is different
```

mik-laj commented on docs/spelling_wordlist.txt:
```diff
@@ -1094,6 +1094,8 @@ pypa
 pytest
 pythonpath
+python_operator
+PythonOperator
```
```suggestion
```
See: https://github.com/apache/airflow/pull/12872/files#r538215678

vikramkoka committed:
Apply suggestions from code review

Formatting changes to avoid spelling list additions

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>
